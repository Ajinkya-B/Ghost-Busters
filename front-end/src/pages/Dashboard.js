import React from 'react';

const Dashboard = () => {
  return (
    <div
      style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        height: '90vh'
      }}
    >
      <h1>Dashboard</h1>
    </div>
  );
};

export default Dashboard;
