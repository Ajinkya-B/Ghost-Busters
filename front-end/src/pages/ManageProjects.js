// This is the Manage Project page.

import React from 'react';
import AddProject from "../components/AddProject";
import ProjectsList from "../components/ProjectsList";
import Navbar from "../components/Navbar";


export default function ManageProjects() {
    return (
      <div>
          <Navbar />
          <br />

          <h3> Add a new project: </h3>
          <AddProject />
          <br />
          <h3> Select an existing project: </h3>
          <ProjectsList />
      </div>
    );
}
