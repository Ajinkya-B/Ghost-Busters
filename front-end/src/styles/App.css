/* This is the font and format styling file for App.js.

Aspects of this file are derived from Tania Rascia's Primitive UI
(Source: https://github.com/taniarascia/primitive |
Documentation: https://taniarascia.github.io/primitive |
MIT License).

*/

/* Fonts
   ========================================================================== */
@import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');

* {
  font-family: 'Roboto', sans-serif;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}


/* Document
   ========================================================================== */
/**
/**
* 1. Correct the line height in all browsers.
* 2. Prevent adjustments of font size after orientation changes in iOS.
*/
html {
  line-height: 1.15;
  /* 1 */
  -webkit-text-size-adjust: 100%;
  /* 2 */
}


/* Sections
   ========================================================================== */
/**
 * Correct the font size and margin on `h1` elements within `section` and
 * `article` contexts in Chrome, Firefox, and Safari.
 */
h1 {
  font-size: 2em;
  margin: 0.67em 0;
}
/**
 * Correct the font size and margin on `h2` elements within `section` and
 * `article` contexts in Chrome, Firefox, and Safari.
 */
h2 {
  font-size: 2em;
  margin: 0.67em 0;
  padding-left: 15px;
}
/**
 * Correct the font size and margin on `h2` elements within `section` and
 * `article` contexts in Chrome, Firefox, and Safari.
 */
h3 {
  font-size: 2em;
  margin: 0.67em 0;
  padding-left: 15px;
}


/* Forms
   ========================================================================== */
/**
 * Format the form text fields.
 */
[type=text] {
  display: block;
  border: 1px solid #dedede;
  border-radius: 10px;
  padding: 0.65rem;
  outline: none;
  background: transparent;
  margin-bottom: 0.5rem;
  font-size: 1rem;
  width: 50%;
  max-width: 50%;
  line-height: 1;
}
label {
  font-weight: 600;
  max-width: 100%;
  display: block;
  margin: 1rem 0 0.5rem;
  padding-left: 30px;
}


/* Buttons
   ========================================================================== */
counterButton {
  cursor: pointer;
}


/* Misc
 ========================================================================== */
/**
 * Scaffolding
 */
html {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font: normal normal normal 1rem/1.6 -apple-system, BlinkMacSystemFont, Helvetica Neue, Helvetica, Arial, sans-serif;
}

body {
  color: #404040;
  background: white;
  font-size: 1rem;
}

h1 {
  font-size: 1.75rem;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 1.25rem;
}

h4 {
  font-size: 1.1rem;
}

h5 {
  font-size: 1rem;
}

/**
 * Extra space for the dashboard UI
 */
.footer { margin-top: 200px; }
