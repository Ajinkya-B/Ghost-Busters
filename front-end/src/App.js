import React, {Component} from 'react';
import './styles/App.css';
import './styles/index2.css';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import "bootstrap/dist/css/bootstrap.min.css";

import Navbar from './components/Navbar';

import Dashboard from './pages/Dashboard';
import ManageProject from './pages/ManageProject';
import AnalyzeTranscripts from './pages/AnalyzeTranscripts';
import Project from './components/Project';


class App extends Component {
  render () {
    return (
      <div>
        <Router>
          <Navbar />
          <br />
          <Routes>
            <Route exact path='/Dashboard/:id' element={< Dashboard />} />
            <Route exact path='/ManageProject' element={< ManageProject />} />
            <Route  path='/projects/:id' element={<Project />} />

            <Route path='/projects/:id/AnalyzeTranscripts' element={< AnalyzeTranscripts />} />
          </Routes>
        </Router>
      </div>


    );
  }
}

export default App;
