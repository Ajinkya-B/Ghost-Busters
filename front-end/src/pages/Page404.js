import React from "react";
import { Helmet } from 'react-helmet-async';
import { Link as RouterLink } from 'react-router-dom';
// @mui
import { styled } from '@mui/material/styles';
import { Button, Typography, Container, Box } from '@mui/material';
// images
import ghost_404 from "../images/ghost-404.png"

// ----------------------------------------------------------------------

const StyledContent = styled('div')(({ theme }) => ({
    maxWidth: 480,
    margin: 'auto',
    minHeight: '100vh',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    padding: theme.spacing(12, 0),
}));

// ----------------------------------------------------------------------

export default function Page404() {
    return (
        <>
            <Helmet>
                <title> 404 Page Not Found </title>
            </Helmet>

            <Container>
                <StyledContent sx={{ textAlign: 'center', alignItems: 'center' }}>
                    <Typography variant="h3" paragraph>
                        Sorry, page not found!
                    </Typography>

                    <Typography sx={{ color: 'text.secondary' }}>
                        Our application never wants to ghost you!
                        <br />
                        Perhaps youâ€™ve mistyped the URL?
                    </Typography>

                    <Box
                        component="img"
                        src={ghost_404}
                        sx={{ height: 250, my: { sm: 5 } }}
                    />

                    <br />

                    <Button to="/" size="large" variant="contained" component={RouterLink}>
                        Go Home
                    </Button>
                </StyledContent>
            </Container>
        </>
    );
}
