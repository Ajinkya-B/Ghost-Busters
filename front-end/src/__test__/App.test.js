/**
 * @jest-environment jsdom
 */

import React from "react";
import {create} from "react-test-renderer";
import {MemoryRouter, Route, Router, Routes} from "react-router-dom";
import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import ManageProjects from "../pages/ManageProjects.js";
import Page404 from "../pages/Page404.js";
import DashboardWelcomePage from "../pages/DashboardWelcome.js";
import Dashboard from "../pages/Dashboard.js";
import App from "../App.js"
import {createMemoryHistory} from "history";

describe('App', () => {
    it("renders the Navbar navigation correctly", async() => {
        let renderer = create(
            <MemoryRouter initialEntries={["/"]}>
                <Routes>
                    <Route path="/ManageProjects" element={<ManageProjects />} />
                    <Route path="*" element={<Page404 />} />
                    <Route path='/' element={<DashboardWelcomePage />} />
                    <Route path='/Dashboard/:id' element={< Dashboard />} />
                </Routes>
            </MemoryRouter>
        );
        expect(renderer.toJSON()).toMatchSnapshot();
    });

    it('renders the Manage Projects page from Home', async() => {
        render(<App />);

        expect(screen.getByRole('link', {name: 'Home'}));

        await userEvent.click(screen.getByRole('link', {name: 'Manage Projects'}))
        expect(screen.getByRole('link', {name: 'Manage Projects'}));
    });

    // Axios error
    it.skip('renders the Home page from the Manage Projects page', async() => {
        render(
            <MemoryRouter initialEntries={['/ManageProjects']}>
                <ManageProjects />
            </MemoryRouter>
        );
        await userEvent.click(screen.getByRole('link', {name: 'Home'}));
        expect(screen.getByRole('link', {name: 'Home'}));
    });

    it.skip('correctly lands on an error page', async() => {
        const badRoute = '/some/bad/route'
        render(
            <MemoryRouter initialEntries={[badRoute]}>
                <ManageProjects />
            </MemoryRouter>
        );
        expect(screen.getByText(/sorry, page not found!/i)).toBeInTheDocument();
    });

});
