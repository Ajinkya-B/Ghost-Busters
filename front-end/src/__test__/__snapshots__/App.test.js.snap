// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the Navbar navigation correctly 1`] = `
<div
  style={
    {
      "background-color": "#f7f7f7",
      "background-image": "url([object Object])",
      "background-position": "left",
      "background-repeat": "no-repeat",
      "height": "100%",
      "position": "fixed",
      "width": "100%",
    }
  }
>
  <nav
    className="sc-bcXHqe wPPAu"
  >
    <a
      aria-current="page"
      className="sc-gswNZR RClpZ active"
      href="/"
      onClick={[Function]}
    >
      <h2>
        Home
      </h2>
    </a>
    <a
      activeStyle={true}
      className="sc-gswNZR RClpZ"
      href="/ManageProjects"
      onClick={[Function]}
    >
      Manage Projects
    </a>
    <nav
      className="sc-dkrFOg jijIET"
    >
      <a
        aria-current="page"
        className="sc-hLBbgP jYsoAB active"
        href="/"
        onClick={[Function]}
      >
        Upload Transcripts
      </a>
    </nav>
  </nav>
  <div
    style={
      {
        "color": "black",
        "padding": "18%",
        "textAlign": "right",
      }
    }
  >
    <h1
      className="h1 title fw-bolder"
    >
      Welcome to your Ghostboard!
    </h1>
    <br />
    <h4
      className="subtitle"
    >
      Please select a project.
    </h4>
    <br />
    <a
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-9y1egq-MuiButtonBase-root-MuiButton-root"
      href="/ManageProjects"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Manage Projects
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
